Young-N-Ambitious/yna_backend
# BA-001.2 get api running

## Fixes
I installed the rack-cors gem, installed devise for authentication, set up the user table with attributes, set up user controller with actions, set up user route resources.
The API is now functioning 
-was able to create a dummy user in the console and get the user to show in json format in the browser. 

Test my api:
-you may need to update the ruby version depending on your environment 
- run ```rvm install "ruby-3.0.0"``` (optional if error occurs... est. 5 mins)
- run ```bundle install``` in your terminal to install your dependencies
- run ```rails db:migrate``` to migrate your database

create a dummy user:
- run ```rails c``` in your terminal to open up your console
- run ```user = User.new``` to create a new user
- run ```user.email = "test@gmail.com"``` to create the user email (do this for all the remaining attributes as well... email and password fields are currently required)
- run ```user.save``` to save your user to the database
- run ```User.all``` to see if your user has been created successfully
- type ```exit``` to leave the rails console

start the server:
-run ```rails s``` to start the server (you should see the rails landing page on successful load)
-navigate to the route /api/V1/users to see your created users 

Test in GitPod:
https://youngnambiti-ynabackend-i5zrnxe8sgk.ws-us90.gitpod.io/
Test API URL after starting the server on gitpod:
https://3000-youngnambiti-ynabackend-i5zrnxe8sgk.ws-us90.gitpod.io/api/v1/users

If everything works correctly... You should see your list of users on the browser in json format.

This screenshot below is a sample user with all the fields completed. This view is from the gitpod workspace.


![Screen Shot 2023-03-10 at 7 25 58 PM](https://user-images.githubusercontent.com/30539832/224462537-7b3657a6-ca08-4535-8056-ef5f02ed920b.png)




Fixes 
Issue Title: BA-0001.2 - Get API Running
Github #[BA-0001.2]
Created by @[@Sydnee510 ]

## Description



## Testing Instructions


## Checklist

- [x ] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [ x] My pull request targets the *default* branch of the repository (`development`)
- [x ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines




Release Notes
-------------




Decisions
---------



Technical Details
-----------------




Links
-----



## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
